From e346e25001d2fdce7dbc5e2750a341aee9498358 Mon Sep 17 00:00:00 2001
From: ilya guterman <amfernusus@gmail.com>
Date: Mon, 28 Apr 2025 05:46:17 +0900
Subject: [PATCH] fix incompatible pointer warnings
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

prog/sensord/rrd.c:458:42: error: passing argument 2 of ‘rrd_update’ from incompatible pointer type [-Wincompatible-pointer-types]
  458 |                 if ((ret = rrd_update(3, (char **) /* WEAK */ argv))) {
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                                          |
      |                                          char **
/usr/include/rrd.h:179:5: note: expected ‘const char **’ but argument is of type ‘char **’
---
 prog/sensord/rrd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/prog/sensord/rrd.c b/prog/sensord/rrd.c
index a531362b..36d685fc 100644
--- a/prog/sensord/rrd.c
+++ b/prog/sensord/rrd.c
@@ -299,7 +299,7 @@ int rrdInit(void)
 		argv[argc++] = rraBuff;
 		argv[argc] = NULL;
 
-		ret = rrd_create(argc, (char**) argv);
+		ret = rrd_create(argc, (const char**) argv);
 		if (ret == -1) {
 			sensorLog(LOG_ERR, "Error creating RRD file: %s: %s",
 				  sensord_args.rrdFile, rrd_get_error());
@@ -455,7 +455,7 @@ int rrdUpdate(void)
 		const char *argv[] = {
 			"sensord", sensord_args.rrdFile, rrdBuff, NULL
 		};
-		if ((ret = rrd_update(3, (char **) /* WEAK */ argv))) {
+		if ((ret = rrd_update(3, (const char **) /* WEAK */ argv))) {
 			sensorLog(LOG_ERR, "Error updating RRD file: %s: %s",
 				  sensord_args.rrdFile, rrd_get_error());
 		}
-- 
2.49.0

